stacks_queues/queue.py

```diff
@@ -13,49 +13,78 @@ def __init__(self):
         self.store = [None] * INITIAL_QUEUE_SIZE
         self.buffer_size = INITIAL_QUEUE_SIZE
         self.front = -1
-        self.rear = -1
+        self.rear = 0
         self.size = 0
-      
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.front == -1:
+            self.front = 0
+            self.rear = 0
+        elif self.size == self.buffer_size:
+            raise QueueFullException
+
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.front == -1:
+            raise QueueEmptyException
+        else:
+            value = self.store[self.front]
+            self.front = (self.front + 1) % self.buffer_size
+            self.size -= 1
+        if self.front == self.rear:
+            self.front = -1
+            self.rear = -1
+        return value
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        if self.front == -1:
+            return None
+        return self.store[self.front]
         
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        if self.front == -1:
+            return 0
+        return self.size
```

Comment:
Since `size` should already be 0 if the queue is empty, you can probably delete lines 65 - 66. 
```suggestion
        return self.size
```